src/mmrm.cpp

```diff
@@ -156,6 +147,21 @@ Type objective_function<Type>::operator() ()
   matrix<Type> beta_vcov = XtWX_decomposition.solve(Identity);
   REPORT(beta_vcov);
 
+  std::vector<int> visit(n_visits);
+  std::iota(std::begin(visit), std::end(visit), 0);
+  matrix<Type> dist(2, 2);
+  dist << 0, 1, 1, 0;
+  int dim;
+  if (is_spatial) {
+    dim = 2;
+  } else {
+    dim = n_visits;
+  }
+  matrix<Type> covariance_lower_chol = matrix<Type>::Zero(dim * n_groups, dim);
+  for (int r = 0; r < n_groups; r++) {
+    covariance_lower_chol.block(r * dim, 0, dim, dim) = chols_by_group[r]->get_chol(visit, dist);
+    delete chols_by_group[r];
+  }
```

Comment:
this code looks a bit low-level in this context, any chance to move this to a function?

Reply:
yes this part will be refined later. here we are trying to create "full_chol" and return this. However later we should return all subject (and actually, we only save the list of valid visits, not replication of all subjects, and we can easily obtain the chol for each subject depending on the visits)